libs/pbd/pthread_utils.cc

```diff
@@ -23,7 +23,8 @@
 #include <stdint.h>
 #include <string>
 
-#ifndef PLATFORM_WINDOWS
+#if !defined PLATFORM_WINDOWS && defined __GLIBC__
+#include <climits>
```

Comment:
interesting, why is `climits` needed?

Reply:
Without it there's potential for silent future breakage.

`climits` wraps `limits.h`, which defines `PTHREAD_STACK_MIN` [1] (used by `pbd_stack_size`). Currently `PTHREAD_STACK_MIN` is only defined by accident because one of the headers included via `pdb/thread_utils.h` includes `limits.h` somewhere. If any of these headers change, the `PTHREAD_STACK_MIN` definition could disappear, unless we include `climits`.

To illustrate, this snippet only includes the system headers directly included by `pthread_utils.{cc,h}`. It doesn't compile for me:
```cpp
#define _GNU_SOURCE
#include <pthread.h>
#include <signal.h>
#include <string>
#include <stdint.h>
#include <cstring>
#include <set>
#include <dlfcn.h>

#ifndef PTHREAD_STACK_MIN
#error PTHREAD_STACK_MIN not defined
#endif
```
[1] https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/limits.h.html